src/msw/dirdlg.cpp

```diff
@@ -309,39 +259,94 @@ int wxDirDialog::ShowSHBrowseForFolder(WXHWND owner)
 //
 // Returns wxID_OK on success, wxID_CANCEL if cancelled by user or wxID_NONE if
 // an error occurred and we should fall back onto the old dialog.
-#if wxUSE_IFILEDIALOG
+#if wxUSE_IFILEOPENDIALOG
+
+int wxDirDialog::ShowIFileOpenDialog(WXHWND owner)
+{
+    HRESULT hr = S_OK;
+    wxCOMPtr<IFileOpenDialog> fileDialog;
+
+    if ( !InitIFileOpenDialog(m_message, m_path, HasFlag(wxDD_MULTIPLE),
+                              HasFlag(wxDD_SHOW_HIDDEN), fileDialog) )
+    {
+        return wxID_NONE; // Failed to initialize the dialog
+    }
+
+    hr = fileDialog->Show(owner);
+    if ( FAILED(hr) )
+    {
+        if ( hr == HRESULT_FROM_WIN32(ERROR_CANCELLED) )
+        {
+            return wxID_CANCEL; // the user cancelled the dialog
+        }
+        else
+        {
+            wxLogApiError(wxS("IFileDialog::Show"), hr);
+        }
+    }
+    else
+    if ( GetPathsFromIFileOpenDialog(fileDialog, HasFlag(wxDD_MULTIPLE),
+                                     m_paths) )
+    {
+        if ( !HasFlag(wxDD_MULTIPLE) )
+        {
+            m_path = m_paths.front();
+        }
+
+        return wxID_OK;
+    }
+
+    // Failed to show the dialog or obtain the selected folders(s)
+    wxLogSysError(_("Couldn't obtain folder name"), hr);
```

Comment:
The error message doesn't seem ideal, but I admit I have trouble finding better suggestions. It probably doesn't matter anyhow, but in a highly unlikely event this really happens, it would be nice to somehow link it with `wxDirDialog`, so maybe it should be "Couldn't obtain the result from the directory selection dialog" or something like that?

Reply:
We already talked about it. I have been reluctant to make just a small change when the string then needs to be re-translated in all the languages.

I will change the string to what you suggested, if you think it is for the better.

Reply:
Sorry, I somehow totally failed to see that this was an existing message. Yes, I agree that it's not worth changing the translations for a message that nobody is going to see anyhow. Sorry again for wasting your time.